File: samples/generators/jpa/src/main/javagen/topmodel/jpa/sample/demo/entities/securite/profil/Droit.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.entities.securite.profil;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.annotations.Immutable;

import jakarta.annotation.Generated;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Enumerated;
import jakarta.persistence.EnumType;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;

import topmodel.jpa.sample.demo.enums.securite.profil.DroitCode;

/**
 * Droits de l'application.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
@Entity
@Table(name = "DROIT")
@Immutable
@Cache(usage = CacheConcurrencyStrategy.READ_ONLY)
public class Droit {

	@Transient
	public static final Droit CREATE = new Droit(DroitCode.CREATE);
	@Transient
	public static final Droit DELETE = new Droit(DroitCode.DELETE);
	@Transient
	public static final Droit READ = new Droit(DroitCode.READ);
	@Transient
	public static final Droit UPDATE = new Droit(DroitCode.UPDATE);

	/**
	 * Code du droit.
	 */
	@Id
	@Column(name = "DRO_CODE", nullable = false, length = 10, columnDefinition = "varchar")
	@Enumerated(EnumType.STRING)
	private DroitCode code;

	/**
	 * Libellé du droit.
	 */
	@Column(name = "DRO_LIBELLE", nullable = false, length = 100, columnDefinition = "varchar")
	private String libelle;

	/**
	 * Type de profil pouvant faire l'action.
	 */
	@ManyToOne(fetch = FetchType.LAZY, optional = false, targetEntity = TypeDroit.class)
	@JoinColumn(name = "TDR_CODE", referencedColumnName = "TDR_CODE")
	private TypeDroit typeDroit;

	/**
	 * No arg constructor.
	 */
	public Droit() {
		// No arg constructor
	}

	/**
	 * Enum constructor.
	 * @param code Code dont on veut obtenir l'instance.
	 */
	public Droit(DroitCode code) {
		this.code = code;
		switch(code) {
		case CREATE :
			this.libelle = "securite.profil.droit.values.Create";
			this.typeDroit = TypeDroit.WRITE;
			break;
		case DELETE :
			this.libelle = "securite.profil.droit.values.Delete";
			this.typeDroit = TypeDroit.ADMIN;
			break;
		case READ :
			this.libelle = "securite.profil.droit.values.Read";
			this.typeDroit = TypeDroit.READ;
			break;
		case UPDATE :
			this.libelle = "securite.profil.droit.values.Update";
			this.typeDroit = TypeDroit.WRITE;
			break;
		}
	}

	/**
	 * Getter for code.
	 *
	 * @return value of {@link topmodel.jpa.sample.demo.entities.securite.profil.Droit#code code}.
	 */
	public DroitCode getCode() {
		return this.code;
	}

	/**
	 * Getter for libelle.
	 *
	 * @return value of {@link topmodel.jpa.sample.demo.entities.securite.profil.Droit#libelle libelle}.
	 */
	public String getLibelle() {
		return this.libelle;
	}

	/**
	 * Getter for typeDroit.
	 *
	 * @return value of {@link topmodel.jpa.sample.demo.entities.securite.profil.Droit#typeDroit typeDroit}.
	 */
	public TypeDroit getTypeDroit() {
		return this.typeDroit;
	}

	/**
	 * Enumération des champs de la classe {@link topmodel.jpa.sample.demo.entities.securite.profil.Droit Droit}.
	 */
	public enum Fields  {
        CODE(DroitCode.class), //
        LIBELLE(String.class), //
        TYPE_DROIT(TypeDroit.class);

		private Class<?> type;

		private Fields(Class<?> type) {
			this.type = type;
		}

		public Class<?> getType() {
			return this.type;
		}
	}
}

File: samples/generators/php/src/Entity/Securite/Profil/Droit.php
<?php
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

namespace App\Entity\Securite\Profil;

use App\Repository\Securite\Profil\DroitRepository;
use Doctrine\ORM\Mapping\Column;
use Doctrine\ORM\Mapping\Entity;
use Doctrine\ORM\Mapping\Id;
use Doctrine\ORM\Mapping\JoinColumn;
use Doctrine\ORM\Mapping\ManyToOne;
use Doctrine\ORM\Mapping\Table;

#[Entity(repositoryClass: DroitRepository::class)]
#[Table(name: 'DROIT')]
class Droit
{
  #[Id]
  #[Column(name: 'DRO_CODE', length: 10)]
  private string $code;

  #[Column(name: 'DRO_LIBELLE', length: 100)]
  private string $libelle;

  #[ManyToOne(targetEntity: TypeDroit::class)]
  #[JoinColumn(name: 'TDR_CODE', referencedColumnName: 'TDR_CODE')]
  private TypeDroit $typeDroit;

  public function getCode(): string
  {
    return $this->code;
  }

  public function getLibelle(): string
  {
    return $this->libelle;
  }

  public function getTypeDroit(): TypeDroit
  {
    return $this->typeDroit;
  }

  public function setCode(string|null $code): self
  {
    $this->code = $code;

    return $this;
  }

  public function setLibelle(string|null $libelle): self
  {
    $this->libelle = $libelle;

    return $this;
  }

  public function setTypeDroit(TypeDroit|null $typeDroit): self
  {
    $this->typeDroit = $typeDroit;

    return $this;
  }
}
